onfido-java/src/test/java/com/onfido/integration/DocumentManagerTest.java

```diff
@@ -6,164 +6,126 @@
-
-    // Correct response body
-    assertEquals("file.png", document.getFileName());
+    assertEquals("file.png", lookupDocument.getFileName());
   }
 
   @Test
-  public void listDocuments() throws Exception {
-    String response =
-        new JsonObject()
-            .add(
-                "documents",
-                Arrays.asList(
-                    new JsonObject().add("file_name", "file1.png").map,
-                    new JsonObject().add("file_name", "file2.png").map))
-            .toJson();
-
-    MockWebServer server = mockRequestResponse(response);
-
-    Onfido onfido =
-        Onfido.builder().apiToken("token").unknownApiUrl(server.url("/").toString()).build();
-
-    List<Document> documents = onfido.document.list("id");
-
-    // Correct path
-    RecordedRequest request = server.takeRequest();
-    assertEquals("/documents/?applicant_id=id", request.getPath());
-
-    // Correct response body
-    assertEquals("file1.png", documents.get(0).getFileName());
-    assertEquals("file2.png", documents.get(1).getFileName());
-  }
+  public void listDocumentsTest() throws Exception {
+    Document anotherDocument = uploadDocument(applicant, "anotherFile.png");
 
-  @Test
-  public void nullParamRequest() throws Exception {
-    String response = new JsonObject().add("file_name", "file.png").toJson();
+    prepareMock(new JsonObject().add(
+                    "documents",
+                    Arrays.asList(
+                        new JsonObject().add("file_name", "anotherFile.png").map,
+                        new JsonObject().add("file_name", "file.png").map)));
 
-    MockWebServer server = mockRequestResponse(response);
+    List<Document> documents = onfido.document.list(applicant.getId()).stream()
+                                              .sorted(Comparator.comparing(Document::getFileName))
+                                              .collect(Collectors.toList());
 
-    Onfido onfido =
-        Onfido.builder().apiToken("token").unknownApiUrl(server.url("/").toString()).build();
+    takeRequest("/documents/?applicant_id=" + applicant.getId());
+
+    assertEquals("anotherFile.png", documents.get(0).getFileName());
+    assertEquals("file.png", documents.get(1).getFileName());
+  }
+
+  @Test
+  public void nullParamRequestTest() throws Exception {
+    prepareMock("error", null, 400);
 
     InputStream inputStream = new ByteArrayInputStream("testing testing 1 2".getBytes());
-    Document.Request documentRequest = Document.request();
-    Document document = onfido.document.upload(inputStream, "file.png", documentRequest);
 
-    // Correct path
-    RecordedRequest request = server.takeRequest();
-    assertEquals("/documents/", request.getPath());
+    try {
+      // uploadDocument(Document.request(), filename);
 
-    // Correct response body
-    assertEquals("file.png", document.getFileName());
+      onfido.document.upload(inputStream, "file.png", Document.request());
+      Assert.fail();
```

Comment:
This assertion is making the test fail. Is it necessary when we're mocking the response to be a failure? 🤔